### ISSUE TITLE
Error Domain=SDWebImageErrorDomain Code=2001 "(null)" UserInfo={SDWebImageErrorDownloadStatusCodeKey=404}

### ISSUE THREAD

**Issue by Zhangyunjiang123** [SDWebImage/SDWebImage#2943]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos / tvos / watchos
 Platform Version        | e.g. 12.0 / 10.14.0 / 12.0 / 5.0
 SDWebImage Version      | e.g. 5.0.0 / 4.4.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 11 / Xcode 10
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by dreampiggy** [comment]
This is just what you see. The server returns 404.
See the documentation again about the error handling.

**Comment by welcome2github** [comment]
@Zhangyunjiang123 请问这个问题你解决了吗，我遇到和你一摸一样的报错信息了

**Comment by dreampiggy** [comment]
HTTP status code 200~299 are considered success, others are all error. You can check the error code to filter the case. 

See Error Code 2001: https://sdwebimage.github.io/SDWebImage/Enums/SDWebImageError.html#/c:@E@SDWebImageError@SDWebImageErrorInvalidDownloadStatusCode

**Comment by dreampiggy** [comment]
Maybe we can provide a description for this error code. Like that `Downloaded image has 0 pixels` for `SDWebImageErrorBadImageData`. TODO in v5.6.0.
